### PR TITLE
Optimize image size

### PR DESCRIPTION
**Pull request by anuraaga**
`RUN chown` touches every file effectively doubling the size of the directory in the Docker image it seems. So we do need to make sure to use docker's native chowning.

Also moves classpath computation to build step since no need to do it every time the container starts

### CODE REVIEW

File: docker/Dockerfile.release [hunk 1]
```diff
@@ -35,6 +35,10 @@ RUN mvn org.apache.maven.plugins:maven-dependency-plugin:get \
 RUN mkdir -p /zipkin && cp ~/.m2/repository/io/zipkin/zipkin-server/${version}/zipkin-server-${version}-exec.jar /zipkin && cd /zipkin && jar xf *.jar && rm *.jar
 RUN mkdir -p /zipkin-slim && cp ~/.m2/repository/io/zipkin/zipkin-server/${version}/zipkin-server-${version}-slim.jar /zipkin-slim && cd /zipkin-slim && jar xf *.jar && rm *.jar
 
+# statically evaluate classpath to avoid https://github.com/docker/for-mac/issues/3643
+RUN cd /zipkin && echo .:$(ls ${PWD}/BOOT-INF/lib/*.jar|tr '\n' ':')${PWD}/BOOT-INF/classes > classpath
+RUN cd /zipkin-slim && echo .:$(ls ${PWD}/BOOT-INF/lib/*.jar|tr '\n' ':')${PWD}/BOOT-INF/classes > classpath
```
Comment (codefromthecrypt) [inline docker/Dockerfile.release]: hmm I think this won't be correct, because PWD will be /zipkin-slim where we want it to end up /zipkin later..

Comment (anuraaga) [reply docker/Dockerfile.release]: Oops - thanks for noticing I only tested fat version ;) Fixed to prepend `/zipkin/` to relative paths

**Comment by anuraaga** [comment]
Actually I found out how to use wildcard - perhaps when you tried before you added `.jar` which doesn't seem to be allowed. How about leaving it to Java?

File: docker/Dockerfile.release (hunk 1 above)
Comment (codefromthecrypt) [reply docker/Dockerfile.release]: hey hey hey... not slim != fat.. more like slim++

File: docker/zipkin/run.sh [hunk 2]
```diff
@@ -20,7 +20,7 @@ fi
 # Use main class directly if there are no modules, as it measured 14% faster from JVM running to available
 # verses PropertiesLauncher when using Zipkin was based on Spring Boot 2.1
 if [[ -z "$MODULE_OPTS" ]]; then
-  exec java ${JAVA_OPTS} -cp $(cat /zipkin/classpath) zipkin.server.ZipkinServer
+  exec java ${JAVA_OPTS} -cp '.:BOOT-INF/lib/*:BOOT-INF/classes' zipkin.server.ZipkinServer
```
Comment (codefromthecrypt) [inline docker/zipkin/run.sh:23]: this actually works? wow.. must have fat-fingered something (pun intended)

**Review by codefromthecrypt — APPROVED** [review]

**Pull request merged**